File: 프로그래머스/lv1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.kt
class Solution {
    fun solution(left: Int, right: Int): Int {

        var sum = 0

        for(i in left..right) {
            val arr = Array(i+1) { true }
            var num = 2
            while (num < i) {
                if(i % num != 0) {
                    var j = num
                    while (j < i) {
                        arr[j] = false
                        j += j
                    }
                }
                num++
            }
            val size = arr.filter { it }.size - 1
            if(size%2 == 0) sum += i else sum -= i
        }
        return sum
    }
}

File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.swift
func solution(_ left:Int, _ right:Int) -> Int {
    var answer = 0
    
    for i in left...right {
        var count = 0
        
        for j in 1...i {
            if i % j == 0 {
                count += 1
            }
        }
        
        answer += (count % 2 == 0) ? i : -i
    }
    
    return answer
}